fml/memory/thread_checker.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// A class for checking that the current thread is/isn't the same as an initial
+// thread.
+
+#ifndef LIB_FXL_SYNCHRONIZATION_THREAD_CHECKER_H_
+#define LIB_FXL_SYNCHRONIZATION_THREAD_CHECKER_H_
```

Comment:
Are we worried that someone will pull in both this and the original? (re: the header guard name)

Reply:
Fixed.